From a7c2bfc24b3f8800bf88d0a1446a3f27a9f3cbe5 Mon Sep 17 00:00:00 2001
From: SuperKali <hello@superkali.me>
Date: Sat, 29 Aug 2026 14:30:35 +0200
Subject: [PATCH] arm64: dts: qcom: cq8725s-ayn: drive the fan from the thermal
 zones

The fan node has cooling levels but no thermal trips wired to them, so
the PWM sits at the boot default and the CPU throttles with the fan
barely spinning. Give the four cpuss and the eight gpuss zones active
trips from 40 to 80 degrees with 5 degrees of hysteresis, each stepping
the fan one level up the cooling-levels table.

Signed-off-by: SuperKali <hello@superkali.me>
---
 .../boot/dts/qcom/cq8725s-ayn-common.dtsi     | 267 ++++++++++++++++++
 1 file changed, 267 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/cq8725s-ayn-common.dtsi b/arch/arm64/boot/dts/qcom/cq8725s-ayn-common.dtsi
index 4c0ee61d1478..0cee0f54b98b 100644
--- a/arch/arm64/boot/dts/qcom/cq8725s-ayn-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/cq8725s-ayn-common.dtsi
@@ -1470,3 +1470,270 @@ rtc_offset: rtc-offset@72 {
 		reg = <0x72 0x4>;
 	};
 };
+
+#define FAN_HYSTERESIS 5000
+
+&{/thermal-zones/cpuss-0-0-thermal} {
+	trips {
+		cpu00_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu00_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu00_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu00_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu00_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu00_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu00_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&cpu00_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&cpu00_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&cpu00_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&cpu00_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&cpu00_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&cpu00_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&cpu00_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/cpuss-0-1-thermal} {
+	trips {
+		cpu01_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu01_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu01_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu01_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu01_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu01_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu01_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&cpu01_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&cpu01_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&cpu01_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&cpu01_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&cpu01_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&cpu01_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&cpu01_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/cpuss-1-0-thermal} {
+	trips {
+		cpu10_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu10_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu10_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu10_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu10_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu10_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu10_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&cpu10_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&cpu10_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&cpu10_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&cpu10_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&cpu10_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&cpu10_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&cpu10_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/cpuss-1-1-thermal} {
+	trips {
+		cpu11_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu11_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu11_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu11_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu11_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu11_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		cpu11_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&cpu11_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&cpu11_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&cpu11_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&cpu11_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&cpu11_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&cpu11_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&cpu11_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/gpuss0-thermal} {
+	trips {
+		gpu0_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu0_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu0_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu0_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu0_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu0_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu0_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&gpu0_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&gpu0_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&gpu0_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&gpu0_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&gpu0_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&gpu0_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&gpu0_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/gpuss1-thermal} {
+	trips {
+		gpu1_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu1_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu1_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu1_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu1_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu1_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu1_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&gpu1_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&gpu1_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&gpu1_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&gpu1_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&gpu1_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&gpu1_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&gpu1_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/gpuss2-thermal} {
+	trips {
+		gpu2_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu2_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu2_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu2_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu2_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu2_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu2_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&gpu2_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&gpu2_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&gpu2_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&gpu2_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&gpu2_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&gpu2_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&gpu2_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/gpuss3-thermal} {
+	trips {
+		gpu3_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu3_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu3_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu3_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu3_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu3_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu3_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&gpu3_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&gpu3_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&gpu3_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&gpu3_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&gpu3_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&gpu3_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&gpu3_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/gpuss4-thermal} {
+	trips {
+		gpu4_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu4_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu4_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu4_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu4_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu4_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu4_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&gpu4_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&gpu4_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&gpu4_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&gpu4_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&gpu4_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&gpu4_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&gpu4_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/gpuss5-thermal} {
+	trips {
+		gpu5_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu5_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu5_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu5_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu5_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu5_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu5_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&gpu5_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&gpu5_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&gpu5_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&gpu5_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&gpu5_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&gpu5_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&gpu5_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/gpuss6-thermal} {
+	trips {
+		gpu6_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu6_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu6_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu6_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu6_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu6_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu6_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&gpu6_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&gpu6_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&gpu6_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&gpu6_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&gpu6_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&gpu6_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&gpu6_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
+&{/thermal-zones/gpuss7-thermal} {
+	trips {
+		gpu7_fan0: trip-point-fan0 { temperature = <40000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu7_fan1: trip-point-fan1 { temperature = <50000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu7_fan2: trip-point-fan2 { temperature = <60000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu7_fan3: trip-point-fan3 { temperature = <65000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu7_fan4: trip-point-fan4 { temperature = <70000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu7_fan5: trip-point-fan5 { temperature = <75000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+		gpu7_fan6: trip-point-fan6 { temperature = <80000>; hysteresis = <FAN_HYSTERESIS>; type = "active"; };
+	};
+
+	cooling-maps {
+		map0 { trip = <&gpu7_fan0>; cooling-device = <&pwm_fan 0 1>; };
+		map1 { trip = <&gpu7_fan1>; cooling-device = <&pwm_fan 1 2>; };
+		map2 { trip = <&gpu7_fan2>; cooling-device = <&pwm_fan 2 3>; };
+		map3 { trip = <&gpu7_fan3>; cooling-device = <&pwm_fan 3 4>; };
+		map4 { trip = <&gpu7_fan4>; cooling-device = <&pwm_fan 4 5>; };
+		map5 { trip = <&gpu7_fan5>; cooling-device = <&pwm_fan 5 6>; };
+		map6 { trip = <&gpu7_fan6>; cooling-device = <&pwm_fan 6 7>; };
+	};
+};
+
-- 
2.47.3

